### PR TITLE
IOS-XR ospf factor out area num or ip into ospf_area

### PR DESCRIPTION
**Pull request by arifogel**

### CODE REVIEW

File: projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java [hunk 1]
```diff
@@ -7606,14 +7600,10 @@ private static long toLong(Literal_communityContext ctx) {
...
-    try {
-      long val = Long.parseLong(ctx.getText(), 10);
-      checkArgument(0 <= val && val <= 0xFFFFFFFFL);
-      return val;
-    } catch (IllegalArgumentException e) {
-      return convProblem(Long.class, ctx, null);
-    }
+  private static long toLong(Uint32Context ctx) {
+    Long val = Longs.tryParse(ctx.getText());
+    assert val != null;
```
Comment (dhalperi) [inline projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java:7605]: How can this be needed? It's `uint32`.

**Review by dhalperi — APPROVED** [review]

**Comment by arifogel** [comment]
I guess it isn't, since it will be unboxed anyway.

**Review by dhalperi — COMMENTED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6924)* status: all files reviewed, 1 unresolved discussion (waiting on @arifogel)

**Review by arifogel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6924)* status: all files reviewed, 1 unresolved discussion (waiting on @dhalperi)

___
*[projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java, line 7605 at r1](https://reviewable.io/reviews/batfish/batfish/6924#-MYS2Q4D2UyABu-yRAQ1-r1-7605:-MYS4Kyd6hVNkifdDW_v:bveokdy) ([raw file](https://github.com/batfish/batfish/blob/cbe374acb387526cfc3ef4048a722ec7f71f278d/projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java#L7605)):*
(Repeating for reviewable)

> I guess it isn't, since it will be unboxed anyway.

**Comment by dhalperi** [comment]
___
*[projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java, line 7605 at r1](https://reviewable.io/reviews/batfish/batfish/6924#-MYS2Q4D2UyABu-yRAQ1-r1-7605:-MYS4hZ4276nLFYOVkhP:b3dpby3) ([raw file](https://github.com/batfish/batfish/blob/cbe374acb387526cfc3ef4048a722ec7f71f278d/projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java#L7605)):*
<details><summary><i>Previously, arifogel (Ari Fogel) wrote…</i></summary><blockquote>

(Repeating for reviewable)

> I guess it isn't, since it will be unboxed anyway.

</blockquote></details>

I was wondering why you weren't using [`long Long.parseLong`](https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#parseLong(java.lang.String)). Why `tryParse` at all?

**Review by arifogel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6924)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @dhalperi)

___
*[projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java, line 7605 at r1](https://reviewable.io/reviews/batfish/batfish/6924#-MYS2Q4D2UyABu-yRAQ1-r1-7605:-MYS48Vc1dDpB_tN9Vya:b-g4tya8) ([raw file](https://github.com/batfish/batfish/blob/cbe374acb387526cfc3ef4048a722ec7f71f278d/projects/batfish/src/main/java/org/batfish/grammar/cisco_xr/CiscoXrControlPlaneExtractor.java#L7605)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

I was wondering why you weren't using [`long Long.parseLong`](https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#parseLong(java.lang.String)). Why `tryParse` at all?
</blockquote></details>

got it. changed.

**Review by dhalperi — COMMENTED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6924)* status: all files reviewed, 1 unresolved discussion (waiting on @dhalperi)

**Pull request merged**